comm/src/main/scala/coop/rchain/comm/discovery/GrpcKademliaRPC.scala

```diff
@@ -31,23 +31,30 @@ class GrpcKademliaRPC(timeout: FiniteDuration)(
     for {
       _     <- Metrics[Task].incrementCounter("ping")
       local <- peerNodeAsk.ask
-      ping  = Ping().withSender(toNode(local))
+      ping  = Ping().withSender(toNode(local)).withNetworkId(networkId)
       pongErr <- withClient(peer, timeout)(
                   _.sendPing(ping)
                     .timer("ping-time")
                 ).attempt
-    } yield pongErr.fold(kp(false), kp(true))
+    } yield pongErr.fold(kp(false), _.networkId == networkId)
```

Comment:
So here the "default case" in the fold, the pong fails. And the pong only has succeeded if the pong's network ID is equal to the current's?

Reply:
@KentShikama yes, this is true. The pinged node responses with a `Pong` which contains a network ID. We consider a node as _dead_ if the `Pong` has the wrong network ID. However, please keep in mind that Kademlia pings remote peers only when a bucket full is.